core/tests/puppeteer-acceptance-tests/puppeteer-testing-utilities/user-factory.js

```diff
@@ -0,0 +1,131 @@
+// Copyright 2023 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Utility File for declaring and initializing users.
+ */
+
+let e2eSuperAdmin = e2eBlogAdmin = e2eBlogPostEditor = e2eGuestUser =
+  require('../user-utilities/blog-post-admin-utils.js');
+
+/**
+ * Global user instances that are created and can be reused again.
+ */
+let superAdminInstance = null, blogAdminInstance = null,
```

Comment:
I think only the super admin should be global, right? And in this case this is a tmp super admin that's used only for signing folks up. All others should be local since you might have e.g. two blog editors interacting with each other.

Reply:
we are making the instances global, so if the user already present, then use that else make a new instance of that user. Till now there is no need to have multiple users of same type (except guest user), and if such situation happens in future we can make an array for that user.

Reply:
No please don't do that. Never make things global, where possible. That leads to interference in the tests and it will be hard to maintain.

Have each user be separate, don't reuse them. (The tmp super admin here is a special case: it's "private" and is sort of like a utility for granting roles.)

Reply:
yeah right.. changed the implementation and now there is just super admin a global user. thanks for pointing this.